cmd/samples/pubsub/multireceiver/main.go

```diff
@@ -0,0 +1,89 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"os"
+	"strings"
+
+	"github.com/cloudevents/sdk-go"
+	"github.com/cloudevents/sdk-go/pkg/cloudevents/client"
+	cepubsub "github.com/cloudevents/sdk-go/pkg/cloudevents/transport/pubsub"
+	pscontext "github.com/cloudevents/sdk-go/pkg/cloudevents/transport/pubsub/context"
+	"github.com/kelseyhightower/envconfig"
+)
+
+type envConfig struct {
+	ProjectID string `envconfig:"GOOGLE_CLOUD_PROJECT"`
+
+	SubscriptionIDs string `envconfig:"PUBSUB_SUBSCRIPTIONS" default:"foo" required:"true"`
+}
+
+type Example struct {
+	Sequence int    `json:"id"`
+	Message  string `json:"message"`
+}
+
+func receive(ctx context.Context, event cloudevents.Event, resp *cloudevents.EventResponse) error {
+	fmt.Printf("Event Context: %+v\n", event.Context)
+
+	fmt.Printf("Transport Context: %+v\n", pscontext.TransportContextFrom(ctx))
+
+	data := &Example{}
+	if err := event.DataAs(data); err != nil {
+		fmt.Printf("Got Data Error: %s\n", err.Error())
+	}
+	fmt.Printf("Data: %+v\n", data)
+
+	fmt.Printf("----------------------------\n")
+	return nil
+}
+
+func main() {
+	ctx := context.Background()
+
+	var env envConfig
+	if err := envconfig.Process("", &env); err != nil {
+		log.Printf("[ERROR] Failed to process env var: %s", err)
+		os.Exit(1)
+	}
+
+	var opts []cepubsub.Option
+	opts = append(opts, cepubsub.WithProjectID(env.ProjectID))
+	for _, subscription := range strings.Split(env.SubscriptionIDs, ",") {
+		opts = append(opts, cepubsub.WithSubscriptionID(subscription))
+	}
+
+	t, err := cepubsub.New(context.Background(), opts...)
+	if err != nil {
+		log.Fatalf("failed to create pubsub transport, %s", err.Error())
+	}
+	c, err := client.New(t)
+	if err != nil {
+		log.Fatalf("failed to create client, %s", err.Error())
+	}
+
+	log.Println("Created client, listening...")
+
+	if err := c.StartReceiver(ctx, receive); err != nil {
+		log.Fatalf("failed to start pubsub receiver, %s", err.Error())
+	}
+}
+
+/*
```

Comment:
Should these comments be at the top of the file to describe what it does and how to run it?